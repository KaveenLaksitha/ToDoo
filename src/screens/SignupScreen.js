import { View, Text, Alert, SafeAreaView, StyleSheet, TextInput, Pressable } from 'react-native'
import React, { useState } from 'react'
import { signUp } from '../services/user.services'

const SignupScreen = ({ navigation }) => {

    const [name, setName] = useState("");
    const [email, setEmail] = useState("");
    const [password, setPassword] = useState("");
    const [repassword, setRepassword] = useState("");
    const [age, setAge] = useState(0);

    //method used to signup
    const signupFunc = async () => {
        const payload = {
            name,
            email,
            password,
            age
        }

        if (name != "" && email != "" && password != "" && age != "") {
            if (password === repassword) {
                try {
                    const res = await signUp(payload);
                    if (res.ok) {
                        Alert.alert("Success!")
                        navigation.replace('signin')
                    } else {
                        Alert.alert("Oops! error occured")
                    }
                } catch (err) {
                    Alert.alert("Oops! error occured")
                }
            } else {
                Alert.alert("Passwords are not matching!")
            }
        }
    }

    return (
        <SafeAreaView style={styles.container}>
            <View style={{ marginTop: 10 }}>
                <Text style={styles.text}>Name :</Text>
                <TextInput style={styles.input} placeholder="" onChangeText={(e) => { setName(e) }}></TextInput>
            </View>

            <View style={{ marginTop: 10 }}>
                <Text style={styles.text}>Email :</Text>
                <TextInput style={styles.input} placeholder="" onChangeText={(e) => { setEmail(e) }}></TextInput>
            </View>
            <View style={{ marginTop: 10 }}>
                <Text style={styles.text}>Age : </Text>
                <TextInput style={styles.input} placeholder="" keyboardType="numeric" onChangeText={(e) => { setAge(e) }}></TextInput>
            </View>
            <View style={{ marginTop: 10 }}>
                <Text style={styles.text}>Password :</Text>
                <TextInput style={styles.input} secureTextEntry={true} placeholder="" onChangeText={(e) => { setPassword(e) }}></TextInput>
            </View>
            <View style={{ marginTop: 10 }}>
                <Text style={styles.text}>Re enter the Password :</Text>
                <TextInput style={styles.input} secureTextEntry={true} placeholder="" onChangeText={(e) => { setRepassword(e) }}></TextInput>
            </View>
            <Pressable onPress={() => { signupFunc() }} style={styles.btnSubmit}>
                <Text style={[styles.text, { color: "white" }]}>Signup</Text>
            </Pressable>
        </SafeAreaView>
    )
}

//styles
const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff',
        alignItems: 'center',
        padding: 20
    },
    input: {
        borderWidth: 1,
        borderRadius: 10,
        borderColor: "black",
        width: 350,
        padding: 10,
        color: 'black',
        fontWeight: 'bold',
        fontSize: 16
    },
    text: {
        color: 'black',
        fontSize: 18,
        marginBottom: 5
    },
    btnSubmit: {
        marginTop: 30,
        backgroundColor: "#52007a",
        justifyContent: "center",
        alignItems: "center",
        borderRadius: 15,
        width: "95%",
        padding: 10

    },
});

export default SignupScreen