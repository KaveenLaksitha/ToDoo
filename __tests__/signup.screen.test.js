import React from 'react'
import SignupScreen from '../src/screens/SignupScreen'

import { render, fireEvent } from '@testing-library/react-native'
import renderer from 'react-test-renderer'

describe('Signup Screen', () => {

    it('should render name input in the signup screen', () => {
        const { getByTestId } = render(<SignupScreen />)

        const input = getByTestId('name')
        expect(input).toBeTruthy()
    });

    it('should render email input in the signup screen', () => {
        const { getByTestId } = render(<SignupScreen />)

        const input = getByTestId('email')
        expect(input).toBeTruthy()
    });

    it('should render age input in the signup screen', () => {
        const { getByTestId } = render(<SignupScreen />)

        const input = getByTestId('age')
        expect(input).toBeTruthy()
    });

    it('should render password input in the signup screen', () => {
        const { getByTestId } = render(<SignupScreen />)

        const input = getByTestId('password')
        expect(input).toBeTruthy()
    });

    it('should render re enter password input in the signup screen', () => {
        const { getByTestId } = render(<SignupScreen />)

        const input = getByTestId('re-password')
        expect(input).toBeTruthy()
    });


    it('should render signup button in the signup screen', () => {
        const page = render(<SignupScreen />)

        const signupBtn = page.getByTestId('signupButton')
        expect(signupBtn).toBeTruthy()

    });

    it('should work signup button in the signup screen correctly', () => {
        const { getByTestId } = render(<SignupScreen />)

        const signupBtn = getByTestId('signupButton')
        fireEvent.press(signupBtn)

    });

    test('renders signup page correctly ', () => {
        const tree = renderer.create(<SignupScreen />).toJSON()
        expect(tree).toMatchSnapshot()
    })
});