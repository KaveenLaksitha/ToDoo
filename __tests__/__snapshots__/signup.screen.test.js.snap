// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Screen renders signup page correctly  1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 18,
          "marginBottom": 5,
        }
      }
    >
      Name :
    </Text>
    <TextInput
      onChangeText={[Function]}
      placeholder=""
      style={
        Object {
          "borderColor": "black",
          "borderRadius": 10,
          "borderWidth": 1,
          "color": "black",
          "fontSize": 16,
          "fontWeight": "bold",
          "padding": 10,
          "width": 350,
        }
      }
      testID="name"
    />
  </View>
  <View
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 18,
          "marginBottom": 5,
        }
      }
    >
      Email :
    </Text>
    <TextInput
      onChangeText={[Function]}
      placeholder=""
      style={
        Object {
          "borderColor": "black",
          "borderRadius": 10,
          "borderWidth": 1,
          "color": "black",
          "fontSize": 16,
          "fontWeight": "bold",
          "padding": 10,
          "width": 350,
        }
      }
      testID="email"
    />
  </View>
  <View
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 18,
          "marginBottom": 5,
        }
      }
    >
      Age : 
    </Text>
    <TextInput
      keyboardType="numeric"
      onChangeText={[Function]}
      placeholder=""
      style={
        Object {
          "borderColor": "black",
          "borderRadius": 10,
          "borderWidth": 1,
          "color": "black",
          "fontSize": 16,
          "fontWeight": "bold",
          "padding": 10,
          "width": 350,
        }
      }
      testID="age"
    />
  </View>
  <View
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 18,
          "marginBottom": 5,
        }
      }
    >
      Password :
    </Text>
    <TextInput
      onChangeText={[Function]}
      placeholder=""
      secureTextEntry={true}
      style={
        Object {
          "borderColor": "black",
          "borderRadius": 10,
          "borderWidth": 1,
          "color": "black",
          "fontSize": 16,
          "fontWeight": "bold",
          "padding": 10,
          "width": 350,
        }
      }
      testID="password"
    />
  </View>
  <View
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 18,
          "marginBottom": 5,
        }
      }
    >
      Re enter the Password :
    </Text>
    <TextInput
      onChangeText={[Function]}
      placeholder=""
      secureTextEntry={true}
      style={
        Object {
          "borderColor": "black",
          "borderRadius": 10,
          "borderWidth": 1,
          "color": "black",
          "fontSize": 16,
          "fontWeight": "bold",
          "padding": 10,
          "width": 350,
        }
      }
      testID="re-password"
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#52007a",
        "borderRadius": 15,
        "justifyContent": "center",
        "marginTop": 30,
        "padding": 10,
        "width": "95%",
      }
    }
    testID="signupButton"
  >
    <Text
      style={
        Array [
          Object {
            "color": "black",
            "fontSize": 18,
            "marginBottom": 5,
          },
          Object {
            "color": "white",
          },
        ]
      }
    >
      Signup
    </Text>
  </View>
</RCTSafeAreaView>
`;
